frontend/src/app/features/auth/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,103 @@
+import { AbstractControl, FormControl, FormGroup, ValidatorFn, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { catchError, of } from 'rxjs';
+
+import { AuthService } from 'src/app/api/auth.service';
+import { Component } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { register } from 'src/app/store/actions/login-page.actions';
+
+@Component({
+    selector: 'app-sign-up',
+    templateUrl: './sign-up.component.html',
+    styleUrls: ['./sign-up.component.sass']
+})
+export class SignUpComponent {
+
+
+    public signUpGroup: FormGroup;
+
+    constructor(
+        private authService: AuthService,
+        private store: Store,
+        private route: ActivatedRoute,
+        private router: Router
+    ) {
+        this.signUpGroup = this.buildSignUpGroup();
+    }
+
+    public signUpClick() {
+        this.authService.signUp({
+            firstName: this.signUpGroup.get('firstName')?.value!,
+            lastName: this.signUpGroup.get('lastName')?.value!,
+            email: this.signUpGroup.get('email')?.value!,
+            password: this.signUpGroup.get('password')?.value!
+        }).pipe(
+            catchError((resp) => {
+                if (resp.error === 'Account with this email already exists') {
+                    this.signUpGroup.get('email')?.setErrors({ emailTaken: true });
+                }
+                return of(null);
+            })
+        ).subscribe(
+            (user) => {
+                if (user == null) {
+                    return;
+                }
+                this.store.dispatch(register({ user }));
+                this.router.navigate(['verify-email'], {
+                    relativeTo: this.route.parent
+                });
+            }
+        );
+    }
+
+    private buildSignUpGroup() : FormGroup {
+        const firstNameControl = new FormControl('', [
```

Comment:
I would move validators separately. E.g. in user profile you will have possibility to change user name. It would be bad to define validators there one more time. So you may create some serive(s) with validators that would store common validators.
It is valid for all validators

Reply:
Moved validators to helper in utils folder